pkg/processor/runtime/golang/runtime.go

```diff
@@ -184,15 +185,10 @@ func (g *golang) callEntrypoint(event nuclio.Event, functionLogger logger.Logger
 
 		// calculate how long it took to invoke the function
 		callDuration := time.Since(startTime)
+		responseChan <- processingResult
```

Comment:
Can you explain what you said in the description:
> The problem was that we sometimes attempted to write to the channel when the reader wasn’t ready yet

Is "the channel" the `responseChan` ? because it is created just before the goroutine is run. How can it not be ready?

Reply:
@TomerShor reader is not ready reader becomes ready after we start writer routine

Reply:
in other words, we got to `case result := <-responseChan:` part after a writing in select case with default, which lead to not writing it. So, we need to block on writing.